### DISCUSSION TITLE
I found a bug in isomorphvf2.py

### DISCUSSION THREAD

**Discussion by ffbskt** [networkx/networkx#6516] [Algorithms]
The bug In isomorphvf2.py  match function that does not stop in this code:

`
```
import networkx as nx
from networkx.algorithms import isomorphism
import matplotlib.pyplot as plt

def create_n_pattern(n):
    G = nx.DiGraph()
    G.add_node(1, type="t")
    for obj in range(2, n+2):
      G.add_node(obj, type="eq")
      G.add_edge(1, obj, edge_type='type1', weight=1, time=1)
    return G
g10 = create_n_pattern(10)

def node_match(n1, n2):
    return n1['type'] == n2['type']

G = nx.DiGraph()
for n in [1, 126, 127, 295, 297, 299, 301, 303, 305, 307, 309, 310, 311, 312, 313]:
  G.add_node(n, type='eq')
for e in [(1, 126), (1, 127), (1, 310), (310, 311), (310, 307), (310, 297), (310, 295), (310, 299), (310, 301), (310, 303), (310, 305), (310, 309), (310, 312), (312, 313)]:
  G.add_edge(*e, edge_type='type1', weight=1, time=1)

G.nodes[310]['type'] = 't'
G.nodes[312]['type'] = 't'
G.nodes[1]['type'] = 't'
print(G.nodes[310])
#nx.draw(G)
GM = isomorphism.DiGraphMatcher(G, g10, node_match=node_match)
GM.subgraph_is_isomorphic()
print(GM.mapping)

plt_graph(G)

````

If you want see a graph here it is:
````

def plt_graph(G, labels=False, scale_size=1):
    fig, ax = plt.subplots(figsize=(scale_size * 8, scale_size * 6))
    # Define colors and label positions for each edge type
    colors = {'type1': 'red', 'type2': 'blue', 'type3': 'green'}
    label_pos = {'type1': 0.5, 'type2': 0.35, 'type3': 0.65}

    # Draw the graph
    pos = nx.spring_layout(G)  # Layout for the graph visualization
    nx.draw_networkx_nodes(G, pos, node_color='lightblue')
    for edgetype in set(nx.get_edge_attributes(G, 'edge_type').values()):
        edges = [(u, v) for u, v, e in G.edges(data=True) if e['edge_type'] == edgetype]
        nx.draw_networkx_edges(G, pos, edgelist=edges, edge_color=colors[edgetype], width=2.0, alpha=0.5, 
                              connectionstyle=f"arc3,rad={0.1*(int(edgetype[-1])-1)}", arrows=True)
        if labels:
            edge_labels = {(u, v): d['weight'] for u, v, d in G.edges(data=True) if d['edge_type'] == edgetype}
            nx.draw_networkx_edge_labels(G, pos, edge_labels=edge_labels, label_pos=label_pos[edgetype], font_size=10)
    nx.draw_networkx_labels(G, pos)
    plt.axis('off')
    plt.show()
```
`

**Comment by rossbar** [comment]
It's a little hard to follow the entire example as there's a lot going on, but IIUC `G` is not an isomorphic subgraph of `g10` as the labels for nodes `310` and `312` in `G` were changed to `"t"` and thus do not match the corresponding nodes in `g10`.
